src/sysinfo.cc

```diff
@@ -517,54 +517,8 @@ int GetNumCPUsImpl() {
   }
   return num_cpu;
 #else
-  // Fallback for platforms (such as WASM) that aren't covered above.
-  int num_cpus = 0;
-  int max_id = -1;
-  std::ifstream f("/proc/cpuinfo");
-  if (!f.is_open()) {
-    std::cerr << "Failed to open /proc/cpuinfo\n";
-    return -1;
-  }
-#if defined(__alpha__)
-  const std::string Key = "cpus detected";
-#else
-  const std::string Key = "processor";
-#endif
-  std::string ln;
-  while (std::getline(f, ln)) {
-    if (ln.empty()) continue;
-    std::size_t split_idx = ln.find(':');
-    std::string value;
-#if defined(__s390__)
-    // s390 has another format in /proc/cpuinfo
-    // it needs to be parsed differently
-    if (split_idx != std::string::npos)
-      value = ln.substr(Key.size() + 1, split_idx - Key.size() - 1);
-#else
-    if (split_idx != std::string::npos) value = ln.substr(split_idx + 1);
-#endif
-    if (ln.size() >= Key.size() && ln.compare(0, Key.size(), Key) == 0) {
-      num_cpus++;
-      if (!value.empty()) {
-        const int cur_id = benchmark::stoi(value);
-        max_id = std::max(cur_id, max_id);
-      }
-    }
-  }
-  if (f.bad()) {
-    PrintErrorAndDie("Failure reading /proc/cpuinfo");
-  }
-  if (!f.eof()) {
-    PrintErrorAndDie("Failed to read to end of /proc/cpuinfo");
-  }
-  f.close();
-
-  if ((max_id + 1) != num_cpus) {
-    fprintf(stderr,
-            "CPU ID assignments in /proc/cpuinfo seem messed up."
-            " This is usually caused by a bad BIOS.\n");
-  }
-  return num_cpus;
+  // Fallback, no other API exists.
+  return -1;
```

Comment:
Should this be
```
#error Don't know how to detect CPU count for this platform?
```
?

Reply:
Well the original code would build even if it would not return an appropriate value.

Depends what the intent is.

Reply:
That's what i'm asking, yes. It does seem like 
Actually, for that, we wouldn't need the return:
```
#else
#error Don't know how to detect CPU count for this platform
#endif
  BENCHMARK_UNREACHABLE();
}
```

Reply:
we had this as a "soft error" only because there are so many platforms out there, and this at least returns 1 CPU in the case that we can't identify it (and logs a thing).  i'm not sure we have good enough coverage yet to make it an error but if you all are confident, go for it.

Reply:
It would be a loud compile-time error that can not be missed,
not a run-time error that you might not notice when packaging,
so i think it would be more safe than the current more-than-silent error...

I'd do it. Especially if we are confident that this fallback is no longer needed.

Reply:
what it means is that any clients that need platform support will now find out.  that may be good or bad, but basic platform support will be "return 1" within an appropriate platform define so that's easy enough.

Reply:
Anyways, i've made my point. Deferring to @dmah42.

Reply:
i'm happy to keep the soft error.  i usually prefer hard errors but i don't really want an influx of random platforms needing support for their way to get CPU.

if we think the new version handles all cases (or should handle all cases) then we can switch it.